src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -278,43 +312,39 @@ export const Deposit: React.FC<DepositProps> = ({
             wallet,
           })
 
-          const fees = await adapter.getFeeData({
+          const fees = (await estimateFees({
+            accountId,
+            contractAddress: undefined,
+            assetId,
+            sendMax: false,
+            cryptoAmount: '0',
             to: customTxInput.to,
-            value: customTxInput.value,
-            chainSpecific: {
-              from: fromAccountId(accountId).account,
-              data: customTxInput.data,
-            },
-          })
+            from: fromAccountId(accountId).account,
+            memo: customTxInput.data,
+          })) as FeeDataEstimate<KnownChainIds.EthereumMainnet>
 
           const fastFeeCryptoBaseUnit = fees.fast.txFee
 
-          const fastFeeCryptoPrecision = bnOrZero(
-            bn(fastFeeCryptoBaseUnit).div(bn(10).pow(feeAsset.precision)),
-          )
+          const fastFeeCryptoPrecision = bn(fromBaseUnit(fastFeeCryptoBaseUnit, feeAsset.precision))
```

Comment:
no need to `bn()` and then `return toString()`

Reply:
5fe5cbc9a4831ca92e3fb146292337d8db58f080